src/api/index.yaml

```diff
@@ -47,6 +47,91 @@ definitions:
             description:
               type: string
 
+  GeometryCollection:
+    type: object
+    description: Geojson geometry collection
+    required:
+      - type
+      - geometry
+    externalDocs:
+      url: http://geojson.org/geojson-spec.html#geometrycollection
+    discriminator: type
+    properties:
+      type:
+        type: string
+        description: the geometry type
+        enum:
+          - Point
+          - LineString
+          - Polygon
+          - MultiPoint
+          - MultiLineString
+          - MultiPolygon
+      geometry:
+        type: object
+
+  Feature:
+    type: object
+    description: Geojson Feature
+    required:
+      - type
+      - geometry
+      - properties
+    externalDocs:
+      url: https://tools.ietf.org/html/rfc7946#section-3.2
+    properties:
+      type:
+        type: string
+        enum:
+          - Feature
+      id:
+        type: string
+      geometry:
+        $ref: '#/definitions/GeometryCollection'
+      properties:
+        $ref: '#/definitions/FeatureProperties'
+
+  FeatureProperties:
+    type: object
+    properties:
+      segId:
+        description: |
+          unique identifier of the event segment
+        type: string
+      type:
+        description: |
+          type of the event that the feature represents
+        type: string
+        enum:
+          - fishing
+          - track
+      coordinateProperties:
+        description: |
+          list of timestamps of each segment position
```

Comment:
this is more a set of additional properties for each geometry position, where each additional property is encoded as an array where the first element in that array is the property value for the first position in the geometry and so on. Not only times are encoded, also speeds and courses.

Reply:
Updated description and spec completed added courses and speed